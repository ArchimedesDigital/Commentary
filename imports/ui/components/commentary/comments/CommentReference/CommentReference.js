
const CommentReference = (props) => {
	if (props.referenceWorks && props.referenceWorks.length) {
		return (
			<div className="comment-reference">
				<h4>Secondary Source(s):</h4>
				<span>
					{props.referenceWorks.map((referenceWork, i) => {
						const isLast = (i === props.referenceWorks.length - 1);

						return (
							<span
								key={referenceWork.slug}
								className="referenceWork"
							>
								{isLast ? ' ' : ''}
								<a
									href={`/referenceWorks/${referenceWork.slug}`}
									rel="noopener noreferrer"
									target="_blank"
								>
									{referenceWork.title}{isLast ? '' : ','}
								</a>
							</span>
						);
					})}
				</span>
			</div>
		);
	}
	return null;
};
CommentReference.propTypes = {
	referenceWorks: React.PropTypes.arrayOf(React.PropTypes.shape({
		title: React.PropTypes.string.isRequired,
		slug: React.PropTypes.string.isRequired,
	})),
};
CommentReference.defaultProps = {
	referenceWorks: null,
};

export default CommentReference;
