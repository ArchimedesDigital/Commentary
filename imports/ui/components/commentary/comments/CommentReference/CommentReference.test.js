import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentReference from './CommentReference';

describe('CommentReference', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentReference />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
