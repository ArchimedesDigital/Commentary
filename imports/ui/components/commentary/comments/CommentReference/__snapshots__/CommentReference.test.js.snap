// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentReference renders correctly 1`] = `null`;
